mail/local/mail.go

```diff
@@ -0,0 +1,183 @@
+package local
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+	"github.com/textileio/go-threads/core/thread"
+	"github.com/textileio/textile/cmd"
+)
+
+var (
+	// ErrNotAMailbox indicates the given path is not within a mailbox.
+	ErrNotAMailbox = errors.New("not a mailbox (or any of the parent directories): .textile")
+	// ErrMailboxExists is used during initialization to indicate the path already contains a mailbox.
+	ErrMailboxExists = errors.New("mailbox is already initialized")
+	// ErrIdentityRequired indicates the operation requires a thread Identity but none was given.
+	ErrIdentityRequired = errors.New("thread Identity is required")
+	// ErrAPIKeyRequired indicates the operation requires am API keybut none was given.
+	ErrAPIKeyRequired = errors.New("api key is required")
+
+	flags = map[string]cmd.Flag{
```

Comment:
Like `buck` this could be made into a CLI pretty easily... these are the config flags.